From a559ccf7247bc448ba30b6ebc02a91fa986f146f Mon Sep 17 00:00:00 2001
From: Jan Friesse <jfriesse@redhat.com>
Date: Thu, 8 Nov 2012 13:17:09 +0100
Subject: [PATCH] Handle segfault in backlog_get

If instance->memb_state is not OPERATION or RECOVERY, we was passing
NULL to cs_queue_used call.

Signed-off-by: Jan Friesse <jfriesse@redhat.com>
Reviewed-by: Fabio M. Di Nitto <fdinitto@redhat.com>
(backport from needle)
---
 exec/totemsrp.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/exec/totemsrp.c b/exec/totemsrp.c
index 9a1bb85..0c18493 100644
--- a/exec/totemsrp.c
+++ b/exec/totemsrp.c
@@ -3368,7 +3368,10 @@ static unsigned int backlog_get (struct totemsrp_instance *instance)
 		backlog = cs_queue_used (&instance->retrans_message_queue);
 		queue_use = &instance->retrans_message_queue;
 	}
-	backlog = cs_queue_used (queue_use);
+
+	if (queue_use != NULL) {
+		backlog = cs_queue_used (queue_use);
+	}
 	
 	instance->stats.token[instance->stats.latest_token].backlog_calc = backlog;
 	return (backlog);
-- 
1.7.1

